src/pipe.js

```diff
@@ -30,5 +34,11 @@ var reverse = require('./reverse');
  *      squareThenDoubleThenTriple(5); //=> 150
  */
 module.exports = function pipe() {
-  return compose.apply(this, reverse(arguments));
+  return reduce(function(f, g) {
+    return curryN(f.length + g.length - 1, function() {
+      var fArgs = take(f.length, arguments);
+      var gArgs = drop(f.length, arguments);
+      return g.apply(this, prepend(f.apply(this, fArgs), gArgs));
+    });
+  }, identity, arguments);
```

Comment:
Yeah, I'm concerned over the performance implications of this change (compose has always been pretty slow but this may be a dramatic hit)